src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -2064,30 +2064,36 @@ public async Task SendAsync_No100ContinueReceived_RequestBodySentEventually()
             if (IsCurlHandler) return;
 
             var clientFinished = new TaskCompletionSource<bool>();
-            const string TestString = "test";
+            const string RequestString = "request";
+            const string ResponseString = "response";
 
             await LoopbackServer.CreateClientAndServerAsync(async uri =>
             {
                 using (HttpClient client = CreateHttpClient())
                 {
                     HttpRequestMessage initialMessage = new HttpRequestMessage(HttpMethod.Post, uri) { Version = VersionFromUseHttp2 };
-                    initialMessage.Content = new StringContent(TestString);
+                    initialMessage.Content = new StringContent(RequestString);
                     initialMessage.Headers.ExpectContinue = true;
-                    HttpResponseMessage response = await client.SendAsync(initialMessage);
+                    using (HttpResponseMessage response = await client.SendAsync(initialMessage))
```

Comment:
This version of SendAsync() is already reading the response body anyways and storing it in a buffer.  So, the subsequent code below:

```c#
await response.Content.ReadAsStringAsync();
```

doesn't change anything in terms of timing of what goes on the wire.

If you really want to separate out receiving the response headers from reading the response body, then this .SendAsync() should be changed to:

```c#
client.SendAsync(initialMessage, HttpCompletionOptions.ResponseHeadersRead);
```

Reply:
> doesn't change anything in terms of timing of what goes on the wire

It wasn't intended to.  The relevant changes are in the server half, not the client half.  This change is just about getting the response sent by the server... doesn't matter whether it's buffered or not.